lib/internal/crypto/scrypt.js

```diff
@@ -82,10 +82,16 @@ function check(password, salt, keylen, options, callback) {
   if (options && options !== defaults) {
     if (options.hasOwnProperty('N'))
       N = validateInt32(options.N, 'N', 0, INT_MAX);
+    if (options.hasOwnProperty('cost'))
```

Comment:
If we do go the route of adding these aliases, I'd prefer if these were `else if`s, and the priority were documented and tested. Otherwise, the priority becomes an implementation detail that could subtly break in the future. Throwing is another option if someone were to provide `N` and `cost`, but that might be overkill.

Reply:
Perhaps `process.emitWarning()` if both are provided would be a softer nudge?

Reply:
I agree that a test for the priority would be a good addition here. Will sign off with that added.

Reply:
I’ve went with the throwing option – it’s not as “soft”, but I don’t think using both options something that would intentionally happen in normal userland code.